packages/flutter_tools/test/general.shard/runner/runner_test.dart

```diff
@@ -451,3 +479,34 @@ class WaitingCrashReporter implements CrashReporter {
     return _future;
   }
 }
+
+/// A fake [Analytics] that will be used to test
+/// the --disable-telemetry flag
+class FakeAnalytics extends Fake implements Analytics {
+  bool _fakeTelemetryStatus = true;
+  bool _fakeShowMessage = true;
+
+  @override
+  String get getConsentMessage => 'message';
+
+  @override
+  bool get shouldShowMessage {
+    final bool result = _fakeShowMessage;
+    if (result) {
+      _fakeShowMessage = false;
```

Comment:
Shouldn't we be setting this to false when `clientShowedMessage` is called?

Reply:
Ah good call, fixing now